File: conociendo-a-rust/src/ejemplos_c/ejemplo2.c
#include <stdio.h>
#include <string.h>

struct Libros {
   char  titulo[50];
   char  autor[50];
   int   identificador;
};

void mostrarLibro(struct Libros libro) {

   printf("Libro título: %s \n", libro.titulo);
   printf("Libro autor: %s \n", libro.autor);
   printf("Libro id: %d \n", libro.identificador);
}

void modificarTitulo(struct Libros* libro, char titulo[50]){
    strcpy(libro->titulo, titulo);
}

int main(int argc, char const *argv[])
{
    struct Libros PrimerLibro;
    struct Libros SegundoLibro;

    strcpy(PrimerLibro.titulo, "Don Quijote de la Mancha");
    strcpy(PrimerLibro.autor, "Miguel de Cervantes"); 
    PrimerLibro.identificador = 1;

    modificarTitulo(&PrimerLibro, "Don Quijote de la Mancha - Edición conmemorativa");
    mostrarLibro(PrimerLibro);

    return 0;
}

File: conociendo-a-rust/src/bin/ejemplo2.rs

struct Libros {
    titulo:String,
    autor: String,
    identificador: i32,
}

fn mostrarLibro(libro: &Libros) {
    println!("Libro título: {}", libro.titulo);
    println!("Libro autor: {}", libro.autor);
    println!("Libro id: {}", libro.identificador);
}

fn modificarTitulo(libro: &mut Libros, titulo: String){
    libro.titulo = titulo;
}

fn main() {

    let mut primerLibro: Libros = Libros { titulo: String::from("Don Quijote de la Mancha"), 
                                autor: String::from("Miguel de Cervantes"), 
                                identificador: 1};
    
    modificarTitulo(&mut primerLibro, String::from("Don Quijote de la Mancha - Edición conmemorativa"));
    
    mostrarLibro(&primerLibro);
    
}
